Fix for issue #49 (#68)
canonical/open-documentation-academy: landscape/explanation/glossary/distribution.md

# Distribution

> In Landscape, a distribution refers to a specific release of the Ubuntu operating System. You can reference it with either the code name (for example jammy), the version number such as 22.04.4, or the actual name (Ubuntu) depending on your current use case.

Reviewer: ```suggestion
In Landscape, a **distribution** refers to the Ubuntu operating system. There are different ways to refer to the distribution depending on your use case.
```

Reviewer: Just some rewording. You did a really great job with your research on distributions in Landscape!!!! I wanted to keep this part more general though, since there are some inconsistencies with how the Landscape API refers to distributions (e.g., sometimes it's the version number and sometimes it's the name). Your information isn't necessarily incorrect, though - from a documentation perspective, I just think it'd be better to provide a general explanation here, and then users can dig into their specific use case for those additional details. Let me know if you have any questions on that :smiley:

Author: Right!! I think this is a better approach too. Something for me to consider for future documentation. Thank you!

> Distributions are useful for many reasons in Landscape. Many Landscape APIs require you to define a distribution in order to use them. Distributions are also important if you want to perform an action such as [repository mirroring](https://ubuntu.com/landscape/docs/explanation-about-repository-mirroring).

Reviewer: ```suggestion
There are some Landscape API methods and web portal features that require you to provide the distribution. For example, distributions are important if you want to mirror repositories via the web portal or API. For more information, see an [explanation of repository mirroring](https://ubuntu.com/landscape/docs/explanation-about-repository-mirroring).
```

> ## Landscape APIs That Require a Distribution

Reviewer: ```suggestion
## Example API methods that require a distribution
```

> The following is a list of some Landscape APIs that need a distribution to function properly:

Reviewer: ```suggestion
The following is a list of example Landscape API methods that need a distribution:
```

Reviewer: "Landscape APIs" would refer to different actual APIs (e.g., if we had a different API for admin actions versus customer actions, or whatever), not the different methods (or "endpoints"). I used the term "methods" for consistency here with the relevant API documentation.

We technically do have two APIs now, but it's only because we're transitioning out of our legacy API (the one in this document) to a new REST API. (Just for some extra info!)

> - **Repositories**: The methods under repositories allow you to manage different repositories. Some of these methods include `CreateDistribution`, and `GetDistributions`. In both cases, you reference the distribution by its name. Read [repository methods for more information](https://ubuntu.com/landscape/docs/api-repositories)

Reviewer: ```suggestion
- **Repositories**: These methods allow you to manage different repositories. For more information, see the [repository methods](https://ubuntu.com/landscape/docs/api-repositories)
```

Reviewer: I made this item and the ones below more general and didn't list specific API methods. This is because it's easier to maintain the document if we keep that information in the other page. Users who are interested in learning more about repository API methods can go to that page to read more.

Author: Thank you for pointing this out. I'm learning more ways to maintain a documentation.

- **Reporting Methods**: These methods allow you to do reporting on selections of computers. They usually take the distribution as a query parameter. For example, here's how to use the `GetCSVComplianceData` method:
    ```shell

>     ?action=GetCSVComplianceData&query=distribution:12.04

Reviewer: ```suggestion
```

Reviewer: I removed the examples for similar reasons as I mentioned above - it's just easier to maintain the document if we keep that information on those API document pages. 

You did a good job with your examples though! :smiley:
